connector_api/connection.h

```diff
@@ -48,9 +87,25 @@ namespace sqlpp
       std::string escape(std::string arg);
     };
 
-    class connection : public sqlpp::connection  // this inheritance helps with ADL for dynamic_select, for instance
+    // The base database-specific connection class. Non-pooled and pooled connection classes derive from it
+    class conn_base : public sqlpp::connection  // this inheritance helps with ADL for dynamic_select, for instance
```

Comment:
Can you rename `conn` to `connection`?

Reply:
Sure. Should I rename all conn_xxxxx classes to connection_xxxxx, i.e. the ones in :
sqlpp11/include/sqlpp11/connection.h
sqlpp11/include/sqlpp11/mysql/connection.h
sqlpp11/include/sqlpp11/postgresql/connection.h
sqlpp11/include/sqlpp11/sqlite3/connection.h

Reply:
OK, I renamed:
conn_base -> connection_base
conn_normal -> connection_normal
conn_pooled -> connection_pooled
_conn_base_t -> _connection_base_t
_conn_pooled_t -> _connection_pooled_t
ConnBase -> ConnectionBase

The tests pass normally, the only (small) issue is that now it is more difficult to tell `connection_pooled` from `connection_pool`

Reply:
Thanks!

> the only (small) issue is that now it is more difficult to tell `connection_pooled` from `connection_pool`

We could rename `connection_pooled` to `pooled_connection`. That would be more natural anyways.

WDYT?

Reply:
I think it is OK, but then we should probably rename `connection_normal` to `normal_connection` too?

Reply:
That sounds good to me, yes.

Reply:
OK, I renamed

connection_pooled -> pooled_connection
connection_normal -> normal_connection
_connection_pooled_t -> _pooled_connection_t

tested the PostgreSQL, MySQL and SQLite3 connectors and pushed the update.

Reply:
Great, thanks!